### PR TITLE
Dom/nice bins

### PR DESCRIPTION
**Pull request by domoritz**
This needs https://github.com/vega/vega-parser/issues/59 to limit the number of bins but it already makes the bin bunds look nice again. 

screen shot 2017-09-30 at 11 56 49

### CODE REVIEW

**Comment by domoritz** [comment]
Fix will be coming with net Vega release: https://github.com/vega/vega-parser/issues/59#issuecomment-333328770.

**Comment by domoritz** [comment]
screen shot 2017-10-01 at 09 13 53

**Comment by domoritz** [comment]
Ready

**Comment by kanitw** [comment]
Can you explain more what [this screenshot](https://github.com/vega/vega-lite/pull/3052#issuecomment-333387355) try to depict?  (What's the problem?)

**Comment by domoritz** [comment]
> Can you explain more what this screenshot try to depict? (What's the problem?)

Oh, that shows what it looks like now. No problems.

**Comment by kanitw** [comment]
> Oh, that shows what it looks like now. No problems.

Is that the default without maxbins?  If so, the number of bins looks pretty crazy.

**Comment by domoritz** [comment]
No, it shows that you can increase the number of bins and the axis still looks good.

**Comment by domoritz** [comment]
okay, something is broken

screen shot 2017-10-10 at 14 30 38


```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {"url": "data/movies.json"},
  "mark": "bar",
  "encoding": {
    "x": {
      "bin": true,
      "field": "IMDB_Rating",
      "type": "ordinal"
    },
    "y": {
      "aggregate": "count",
      "type": "quantitative"
    }
  }
}

```

**Comment by domoritz** [comment]
Fixed example. @kanitw

File: examples/compiled/bar_aggregate_count.svg [hunk 1]
```diff
@@ -1 +1 @@
...
...86)" style="font: 10px sans-serif; fill: #000; opacity: 1;">400</text><text text-anchor="end" transform="translate(-7,117.28571428571429)" style="font: 10px sans-serif; fill: #000; opacity: 1;">600</text><text text-anchor="end" transform="translate(-7,88.71428571428572)" style="font: 10px sans-serif; fill: #000; opacity: 1;">800</text><text text-anchor="end" transform="translate(-7,60.14285714285714)" style="font: 10px sans-serif; fill: #000; opacity: 1;">1,000</text><text text-anchor="end" transform="translate(-7,31.571428571428584)" style="font: 10px sans-serif; fill: #000; opacity: 1;">1,200</text><text text-anchor="end" transform="translate(-7,3)" style="font: 10px sans-serif; fill: #000; opacity: 1;">1,400</text></g><g class="mark-rule role-axis-domain" style="pointer-events: none;"><line transform="translate(0,200)" x2="0" y2="-200" style="fill: none; stroke: #888; stroke-width: 1; opacity: 1;"></line></g><g class="mark-text role-axis-title" style="pointer-events: none;"><text text-anchor="middle" transform="translate(-50,100) rotate(-90) translate(0,-2)" style="font: bold 11px sans-serif; fill: #000; opacity: 1;">Number of Records</text></g></g></g></g></g></g></g></g></svg>
```
Comment (kanitw) [inline examples/compiled/bar_aggregate_count.svg]: @domoritz  I wonder if your result is different from mine on this one.

Comment (domoritz) [reply examples/compiled/bar_aggregate_count.svg]: screen shot 2017-10-15 at 18 18 21

**Comment by domoritz** [comment]
Examples of what is better now

screen shot 2017-10-15 at 19 20 57
screen shot 2017-10-15 at 19 18 46

**Pull request merged**